libs/blocks/global-navigation/global-navigation.js

```diff
@@ -41,70 +33,107 @@ function getBlockClasses(className) {
-    this.decorateBlocks();
     this.desktop = window.matchMedia('(min-width: 1200px)');
+    body.querySelectorAll('[class$="-"]').forEach((block) => {
+      const { name, variants } = getBlockClasses(block.className);
+      block.classList.add(name, ...variants);
+    });
   }
 
   init = () => {
-    this.state = {};
-    this.curtain = createTag('div', { class: 'gnav-curtain' });
-    const nav = createTag('nav', { class: 'gnav', 'aria-label': 'Main' });
-
-    const mobileToggle = this.decorateToggle(nav);
-    nav.append(mobileToggle);
-
-    const brand = this.decorateBrand();
-    if (brand) {
-      nav.append(brand);
-    }
-
-    const scrollWrapper = toFragment`
-      <div class="mainnav-wrapper">
-
+    this.curtain = toFragment`<div class="gnav-curtain"></div>`;
+    const nav = toFragment`
+      <div class="gnav-wrapper">
+        <nav class="gnav" aria-label="Main">
+          ${this.mobileToggle()}
+          ${this.decorateBrand()}
+          <div class="mainnav-wrapper">
+            ${this.decorateMainNav()}
+            ${this.decorateSearch()}
+          </div>
+          ${this.decorateProfile()}
+          ${this.decorateLogo()}
+        </nav>
+        ${this.decorateBreadcrumbs()}
       </div>
-    `
-
-    const mainNav = this.decorateMainNav();
-    if (mainNav) {
-      const cta = this.decorateCta();
-      if (cta) {
-        mainNav.append(cta);
-      }
-      scrollWrapper.append(mainNav);
-    }
-
-    const search = this.decorateSearch();
-    if (search) {
-      scrollWrapper.append(search);
-    }
-
-    if (scrollWrapper.children.length > 0) {
-      nav.append(scrollWrapper);
-    }
-
-    const profile = this.decorateProfile();
-    if (profile) {
-      nav.append(profile);
-    }
+    `;
+    this.el.addEventListener('click', this.loadDelayed);
+    setTimeout(() => this.loadDelayed(), 3000);
+    this.el.append(this.curtain, nav);
+  };
 
-    const logo = this.decorateLogo();
-    if (logo) {
-      nav.append(logo);
-    }
+  loadDelayed = async () => {
+    this.ready = this.ready || new Promise(async (resolve) => {
+      this.el.removeEventListener('click', this.loadDelayed);
+      const [
+        { MenuControls },
+        { decorateMenu, decorateLargeMenu },
+        { appLauncher },
+        { profile },
+      ] = await Promise.all([
+        loadBlock('./delayed-utilities.js'),
+        loadBlock('./blocks/navMenu/menu.js'),
+        loadBlock('./blocks/appLauncher/appLauncher.js'),
+        loadBlock('./blocks/profile/profile.js'),
+        loadStyles('navMenu/menu.css'),
+      ]);
+      this.menuControls = new MenuControls();
+      this.decorateMenu = decorateMenu;
+      this.decorateLargeMenu = decorateLargeMenu;
+      this.appLauncher = appLauncher;
+      this.profile = profile;
+
+      this.imsReady
+        .then(({ blockEl, profileEl }) => {
+          this.decorateProfileMenu(blockEl, profileEl);
+        });
+      resolve();
+    });
+    return this.ready;
+  };
```

Comment:
@chrischrischris @auniverseaway would love to hear your opinions on this as we want to put performance on #1 and only ship an initial crucial paint of the nav. Then after 3? seconds or on click load all the interaction logic and initially not visible CSS for the Menu/Search/Profile/etc. (Loading the files should be cached, so interaction should feel seamless to a user - or on mobile after 3s if a user takes his time, he can still open a menu when he loses connection as we preload it)

For now, we just load everything, as the nav grows we might only load the necessary parts or get a bit smarter with the logic, so we don't do 30 requests for 15 different things of which only 3-4 are realistically used.